config_bucket.go

```diff
@@ -0,0 +1,72 @@
+package evergreen
+
+import (
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	BucketTypeLocal = "local"
+	BucketTypeS3    = "s3"
+)
+
+// BucketConfig represents the admin config section for bucket storage.
+type BucketConfig struct {
+	LogBucket     string `bson:"log_bucket" json:"log_bucket" yaml:"log_bucket"`
+	LogBucketType string `bson:"log_bucket_type" json:"log_bucket_type" yaml:"log_bucket_type"`
+}
+
+var (
+	bucketConfigLogBucketKey     = bsonutil.MustHaveTag(BucketConfig{}, "LogBucket")
+	bucketConfigLogBucketTypeKey = bsonutil.MustHaveTag(BucketConfig{}, "LogBucketType")
+)
+
+func (*BucketConfig) SectionId() string { return "bucket" }
+
+func (c *BucketConfig) Get(env Environment) error {
```

Comment:
I changed the interface for Get/Set in my gargantuan PR (https://github.com/evergreen-ci/evergreen/pull/6732).
It should be there if you merge main.

Reply:
Thanks for the head's up!